src/session.rs

```diff
@@ -95,21 +100,30 @@ impl Session {
     /// };
     /// ```
     pub fn new(options: Options) -> AMQPResult<Session> {
-        let connection = try!(get_connection(&options));
+        let connection = get_connection(&options)?;
         let channels = Arc::new(Mutex::new(HashMap::new()));
         let (channel_zero_sender, channel_receiver) = sync_channel(CHANNEL_BUFFER_SIZE); //channel0
         let channel_zero = channel::Channel::new(0, channel_receiver, connection.clone());
-        try!(channels.lock().map_err(|_| AMQPError::SyncError)).insert(0, channel_zero_sender);
+        channels.lock().map_err(|_| AMQPError::SyncError)?.insert(0, channel_zero_sender);
         let con1 = connection.clone();
-        let channels_clone = channels.clone();
+        let channels_clone = Arc::clone(&channels);
         thread::spawn(|| Session::reading_loop(con1, channels_clone));
+
+        // If the caller has configured heartbeats, then begin the heartbeat loop.
+        if options.heartbeat > 0 {
```

Comment:
Rather than checking if heartbeat `> 0`, I think it's preferable to use `Option<T>`.

Reply:
Yea, considered doing that. The reason I went with zero is because, what if a user sets it to `0`? Seems  to be a gray area. Not sure what that is actually supposed to practically mean, other than *hammer the MQ with heartbeats* `:)`. If we go with `Option<u16>` we will just have to cover that edge case and document it in the README & docs.

Thoughts?

Reply:
We could also do something similar with `Duration` where we just check that the duration is some amount of time `>=` 1 second, and only then will we run the heartbeats.